rawsockets doesn't work with cpp backend

```nimrod
import rawsockets
```
Error:
```
/home/deen/git/nimrod-unsorted/nimcache/stdlib_rawsockets.cpp: In function ‘addrinfo* getaddrinfo_142208(NimStringDesc*, NU16, NU8, NU8, NU8)’:
/home/deen/git/nimrod-unsorted/nimcache/stdlib_rawsockets.cpp:579:32: error: invalid conversion from ‘const char*’ to ‘NCSTRING {aka char*}’ [-fpermissive]
   LOC6 = gai_strerror(gairesult);
                                ^
/home/deen/git/nimrod-unsorted/nimcache/stdlib_rawsockets.cpp: In function ‘void gethostbyaddr_143246(NimStringDesc*, thostent141421*)’:
/home/deen/git/nimrod-unsorted/nimcache/stdlib_rawsockets.cpp:787:27: error: invalid conversion from ‘const char*’ to ‘NCSTRING {aka char*}’ [-fpermissive]
   LOC5 = hstrerror(h_errno);
```

Smaller example with the same error:

```nimrod
proc fa4: array[1, bool] = discard
discard fa4()
```
```
/home/deen/git/nim-unsorted/nimcache/min.cpp: In function ‘void minInit()’:
/home/deen/git/nim-unsorted/nimcache/min.cpp:100:16: error: cannot convert ‘bool*’ to ‘bool (*)[1]’ for argument ‘1’ to ‘void fa4_90003(bool (*)[1])’
  fa4_90003(LOC1);
                ^
Error: execution of an external program failed
```

--- a/compiler/cgen.nim
+++ b/compiler/cgen.nim
@@ -486,7 +486,7 @@ proc lenField(p: BProc): PRope =
 include ccgcalls, "ccgstmts.nim", "ccgexprs.nim"
 
 # ----------------------------- dynamic library handling -----------------
-# We don't finalize dynamic libs as this does the OS for us.
+# We don't finalize dynamic libs as the OS does this for us.
 
 proc isGetProcAddr(lib: PLib): bool =
   let n = lib.path

--- a/compiler/installer.ini
+++ b/compiler/installer.ini
@@ -62,7 +62,7 @@ Files: "icons/koch_icon.o"
 
 Files: "compiler/readme.txt"
 Files: "compiler/installer.ini"
-Files: "compiler/nim.nimrod.cfg"
+Files: "compiler/nim.nim.cfg"
 Files: "compiler/*.nim"
 Files: "doc/*.txt"
 Files: "doc/manual/*.txt"

--- a/compiler/main.nim
+++ b/compiler/main.nim
@@ -54,6 +54,7 @@ proc commandDoc2 =
   finishDoc2Pass(gProjectName)
 
 proc commandCompileToC =
+  extccomp.initVars()
   semanticPasses()
   registerPass(cgenPass)
   rodPass()

